### PR TITLE
transientSettings APIs, handle preferences sub-tabs

### LINKED ISSUE #2634: The Preferences Dialog should remember the active tab when it's closed
### Problem Description

Most of the time I want to see which version of K8s I'm running.  But every time I bring up the Preferences box it shows me the Application info, which I actually never care about.

### Proposed Solution

Have the Prefs thing remember the active tab (could use settings.json to store that -- why not?) and bring that up whenever the Prefs dialog is redisplayed.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by torchiaf**
Blocks  #3010

Fixes #2634
Signed-off-by: Francesco Torchia <francesco.torchia@suse.com>

### CODE REVIEW

File: e2e/rdctl.e2e.spec.ts [hunk 1]
```diff
@@ -376,11 +376,45 @@ test.describe('Command server', () => {
...
+    expect(resp2.ok).toBeFalsy();
+    expect(resp2.status).toEqual(400);
+
+    // Now verify that the specified values did not get updated.
+    resp = await doRequest('/v0/transient_settings');
+    const refreshedSettings = await resp.json();
+
+    expect(refreshedSettings).toEqual(transientSettings);
+  });
+
```
Comment (ericpromislow) [inline e2e/rdctl.e2e.spec.ts:412]: I would add tests to verify that you can't update transient settings when you specify a tab for a setting that doesn't support tabs: `'Container Engine'`, `'Kubernetes'`, and `win32 ? 'WSL' : 'Virtual Machine'`

**Review by ericpromislow — CHANGES_REQUESTED** [review]
All this needs are more tests for the non-inner-tab-containing outer-tabs.

File: e2e/rdctl.e2e.spec.ts (hunk 1 above)
Comment (torchiaf) [reply e2e/rdctl.e2e.spec.ts:412]: Sure, thanks Eric, just pushed.

**Review by ericpromislow — APPROVED** [review]
LGTM

**Pull request merged**